### ISSUE TITLE
json output incorrectly inserting arrays surrounding hashes

### ISSUE THREAD

**Issue by liger1978** [hashicorp/terraform#20802]
### Terraform Version
```
Terraform v0.11.13
```

Take this example output spec in the `tf` file:

```hcl
output "test" {
  value = {
    foo = {
      a = "b",
      c = "d",
    }
  }
}
```

Run the command to output as json:

```bash
terraform output -json test
```

Expected output:
```json
{
    "sensitive": false,
    "type": "map",
    "value": {
        "foo": {
            "a": "b",
            "c": "d"
        }
    }
}
```

Actual output:
```
{
    "sensitive": false,
    "type": "map",
    "value": {
        "foo": [
            {
                "a": "b",
                "c": "d"
            }
        ]
    }
}
```

Terraform seems to have inserted an extra array around the elements of the `foo` hash.

**Comment by liger1978** [comment]
Any input on this?

**Comment by apparentlymart** [comment]
Sorry for the late response, @liger1978.

The configuration parser in Terraform 0.11 had some annoying behaviors where it would automatically rewrite a single map as a list of maps in some cases, as a side-effect of Terraform 0.11's poor separation between the idea of a nested block and the idea of a map. (It would internally translate multiple nested blocks into a list of nested blocks, but this internal detail leaked into the interface because it would incorrectly run for nested maps too.)

Terraform 0.12 has fixed this class of issue by handling nested blocks and maps as separate concepts, and thus the new parser will always understand your output value as being a map of map of string, without any extra lists sneaking in.

Because this is fixed in 0.12, I'm going to close this. Unfortunately I think the best workaround we have for 0.11 is to generate the map using the `map` function instead, so that it isn't subject to the block processing of the main language parser:

```hcl
output "test" {
  value = {
    foo = {
      a = "b",
      c = "d",
    }
  }
  value = "${map("foo", map("a", "b", "c", "d"))}"
}
```
